### PR TITLE
stage-3

### PR DESCRIPTION
**Pull request by Merildew**

### CODE REVIEW

File: src/js/headerScript.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+const headerItems = [
```
Comment (ten-ayu) [inline src/js/headerScript.js:1]: Please name constants in `UPPER_CASE`

File: src/js/headerScript.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+function insertHeader(parentNode, className) {
+  headerItems.forEach((item) => {
+    parentNode.append(
+      createHeaderElement(item.title, item.items, ...className)
+    );
+  });
+}
+
+function createHeaderElement(title, items, ...className) {
```
Comment (ten-ayu) [inline src/js/headerScript.js:63]: Looks like `className` should be `classNames`

File: src/styles/mobile-styles.css [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  left: 0;
+  background-color: #191e1e;
+  z-index: 3;
+  padding: 40px 60px;
+  box-sizing: border-box;
+  transform: translateY(-100%);
+  transition: transform 0.35s cubic-bezier(0.694, 0, 0.335, 1);
+}
+
+.nav-list-mobile span,
```
Comment (ten-ayu) [inline src/styles/mobile-styles.css:16]: Please don't use element selectors and avoid nested selectors, plain css class would be almost always better

File: src/js/searchScript.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+const projectList = [
```
Comment (ten-ayu) [inline src/js/searchScript.js:1]: Same, please use UPPER_CASE for constants in the future

Comment (ten-ayu) [inline src/js/searchScript.js:1]: It is good practice to extract constants to separate `constants.js` file

File: src/js/searchScript.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  },
+];
+
+insertResult(projectList);
+
+let typing;
+document
+  .getElementById("searchInput")
+  .addEventListener("keyup", function (event) {
+    clearTimeout(typing);
```
Comment (ten-ayu) [inline src/js/searchScript.js:40]: It is very good that you already know the right way to implement search funnctionallity 
FYI there are already defined patterns for such cases, you can read about `debounce` function 
https://www.youtube.com/watch?v=TBIEArmPywU

**Review by ten-ayu — APPROVED** [review]

File: src/js/headerScript.js (hunk 1 above)
Comment (Qunapi) [inline src/js/headerScript.js:1]: Please name constants in `UPPER_CASE`

File: src/js/headerScript.js (hunk 2 above)
Comment (Qunapi) [inline src/js/headerScript.js:63]: Looks like `className` should be `classNames`

File: src/js/searchScript.js (hunk 4 above)
Comment (Qunapi) [inline src/js/searchScript.js:1]: Same, please use UPPER_CASE for constants in the future

Comment (Qunapi) [inline src/js/searchScript.js:1]: It is good practice to extract constants to separate constants.js file

File: src/js/searchScript.js (hunk 5 above)
Comment (Qunapi) [inline src/js/searchScript.js:40]: It is very good that you already know the right way to implement search funnctionallity
FYI there are already defined patterns for such cases, you can read about `debounce` function
https://www.youtube.com/watch?v=TBIEArmPywU

File: src/styles/mobile-styles.css (hunk 3 above)
Comment (Qunapi) [inline src/styles/mobile-styles.css:16]: Please don't use element selectors and avoid nested selectors, plain css class would be almost always better

**Review by Qunapi — APPROVED** [review]

**Pull request merged**